### PR TITLE
Add new AIMEC plot; add flag for in depth plots; change readASCheader [ana3]

### PR DESCRIPTION
**Pull request by awirb**

### CODE REVIEW

File: avaframe/ana3AIMEC/ana3AIMECCfg.ini [hunk 1]
```diff
@@ -73,6 +73,22 @@ unit =
...
+# if extraPlots true additional analysis plots are created
+extraPlots = True
+# for one to one comparison of two result variables or derived quantities with option to distinguish scenarios using scenario
+# comparison result variables options: resTypeFieldMax (or Min or Mean), maxresTypeCrossMax, sRunout, deltaSXY, zRelease,
+# zRunout, deltaH, relMass, finalMass, entMass, thalwegTravelAngle
+# comparison result variable 1
+compResType1 = pfvFieldMax|deltaSXY
+# comparison result variable 2
+compResType2 = pftFieldMax|runoutAngle
+# scenario parameter name
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMECCfg.ini:86]: What is this parameter for? I think a better description is needed

File: avaframe/ana3AIMEC/ana3AIMECCfg.ini [hunk 2]
```diff
@@ -73,6 +73,22 @@ unit =
 ##relTh = 0.75|0.9|0.8
 
 
+[PLOTS]
+# if extraPlots true additional analysis plots are created
+extraPlots = True
+# for one to one comparison of two result variables or derived quantities with option to distinguish scenarios using scenario
+# comparison result variables options: resTypeFieldMax (or Min or Mean), maxresTypeCrossMax, sRunout, deltaSXY, zRelease,
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMECCfg.ini:80]: We don't have a list what these parameters mean, right? At least not a full one. 
I think the new/missing parameters should be added to the ana3Aimec documentation including their source code names plus the new figures should also be added to the docs

If you don't want to do this with this PR, please add an issue, but start this soon then

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/in2Trans/ascUtils.py [hunk 3]
```diff
@@ -26,59 +26,18 @@ def readASCheader(fname):
...
-        elif ln == 5:
-            if 'nodata' in item[0].lower():
-                headerInfo['nodata_value'] = float(item[1])
-            else:
-                message = ('DEM header is not in correct format - check line 6: should be nodata_value')
-                log.error(message)
-                raise ValueError(message)
-        ln += 1
+                 break
+            row_ite = row_ite+1
```
Comment (fso42) [inline avaframe/in2Trans/ascUtils.py:40]: Please adjust to camelCase (i.e. remove underscores)
And convert ncols and nrows to integer

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/ana3AIMEC/ana3AIMECCfg.ini (hunk 1 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMECCfg.ini]: added more info

File: avaframe/in2Trans/ascUtils.py (hunk 3 above)
Comment (awirb) [reply avaframe/in2Trans/ascUtils.py]: changed

File: avaframe/ana3AIMEC/ana3AIMECCfg.ini (hunk 2 above)
Comment (awirb) [reply avaframe/ana3AIMEC/ana3AIMECCfg.ini:80]: I added an issue #909

File: avaframe/in2Trans/ascUtils.py [hunk 4]
```diff
@@ -26,68 +26,37 @@ def readASCheader(fname):
...
-            if item[0].lower() == 'ncols':
-                headerInfo['ncols'] = int(item[1])
+    # read header
+    headerRows = 6 # six rows for header information
+    headerInfo = {} # store header information including ncols, nrows, xllcorner, yllcorner, cellsize, nodata_value
+    rowCount = 1
+    with open(fname, 'rt') as fileH:
+         for line in fileH:
+            if rowCount <= headerRows:
+                 line = line.split(" ", 1)
```
Comment (fso42) [inline avaframe/in2Trans/ascUtils.py:36]: I think we should stay with the previously used `line.split()`

File: avaframe/out3Plot/plotUtils.py [hunk 5]
```diff
@@ -577,6 +577,8 @@ def putAvaNameOnPlot(ax, avaDir, date=True):
     ax.annotate(
         infoText,
         xy=(0.01, 0.01),
+        color=color,
+        fontsize=fontsize,
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:581]: Is fontsize=None actually the default?

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/out3Plot/plotUtils.py [hunk 6]
```diff
@@ -551,7 +551,7 @@ def addColorBar(
     return cbar
 
 
-def putAvaNameOnPlot(ax, avaDir, date=True):
+def putAvaNameOnPlot(ax, avaDir, date=True, color='k', fontsize=None):
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:554]: add color and fontsize to docstring....

File: avaframe/out3Plot/plotUtils.py (hunk 5 above)
Comment (awirb) [reply avaframe/out3Plot/plotUtils.py:581]: yes but If I want to change it needs to be a parameter as input..

**Review by fso42 — APPROVED** [review]

File: avaframe/out3Plot/plotUtils.py [hunk 7]
```diff
@@ -862,7 +878,23 @@ def fetchContourCoords(xGrid, yGrid, data, level):
...
+        # use codes of path
+        labelID = 0
+        for ind, val in enumerate(contourP.get_paths()[0].codes):
+            if val == 1:
+                labelID = labelID + 1
+                contourDictXY['line%s_%d' % (level, labelID)] = {'x': [contourP.get_paths()[0].vertices[ind, 0]],
+                                                                 'y': [contourP.get_paths()[0].vertices[ind, 1]]}
+            else:
+                contourDictXY['line%s_%d' % (level, labelID)]['x'].append(contourP.get_paths()[0].vertices[ind, 0])
+                contourDictXY['line%s_%d' % (level, labelID)]['y'].append(contourP.get_paths()[0].vertices[ind, 1])
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:899]: Why is this necessary? Is labelID always either 0 or 1? and line 894 and 897 do basically the same?! Or am I missing something?

File: avaframe/out3Plot/plotUtils.py [hunk 8]
```diff
@@ -862,7 +878,23 @@ def fetchContourCoords(xGrid, yGrid, data, level):
     contourP = plt.contour(xGrid, yGrid, data, levels=[level])
 
     contourDictXY = {}
-    for i in range(len(contourP.allsegs[0])):
-        contourDictXY['line%s_%d' %(level, i)] = {'x': contourP.allsegs[0][i][:, 0], 'y':contourP.allsegs[0][i][:, 1]}
+    # loop over all segments of the contour line of the required level
+    # check matplotlib version to fetch coordinates of line
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:882]: Add a quick note why this is needed

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/out3Plot/plotUtils.py (hunk 7 above)
Comment (fso42) [reply avaframe/out3Plot/plotUtils.py:899]: multple lines, val ==1 comes at each start of level

Comment (awirb) [reply avaframe/out3Plot/plotUtils.py:900]: because now all the vertices are provided as one array

**Review by fso42 — APPROVED** [review]

**Comment by fso42** [comment]
Standardtest ok apart from the S/M/L issue
